pkg/antctl/raw/multicluster/get/resourceexport.go

```diff
@@ -0,0 +1,157 @@
+$ antctl mc get resourceexport -A
+Get all ResourceExports in the specified Namespace
+$ antctl mc get resourceexport -n <NAMESPACE>
+Get all ResourceExports and print them in JSON format
+$ antctl mc get resourceexport -o json
+Get the specified ResourceExport
+$ antctl mc get resourceexport <RESOURCEEXPORT> -n <NAMESPACE>
+`, "\n")
+
+func (o *resourceExportOptions) validateAndComplete() {
+	if o.allNamespaces {
+		o.namespace = metav1.NamespaceAll
+		return
+	}
+	if o.namespace == "" {
+		o.namespace = metav1.NamespaceDefault
+		return
+	}
+}
+
+func init() {
+	cmdResourceExport = &cobra.Command{
+		Use: "resourceexport",
+		Aliases: []string{
+			"resourceexports",
+			"re",
+		},
+		Short:   "Print Multi-cluster ResourceExports",
+		Args:    cobra.MaximumNArgs(1),
+		Example: resourceExportExamples,
+		RunE:    runEResourceExport,
+	}
+	o := &resourceExportOptions{}
+	optionsResourceExport = o
+	cmdResourceExport.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Namespace of ResourceExport")
+	cmdResourceExport.Flags().StringVarP(&o.outputFormat, "output", "o", "", "Output format. Supported formats: json|yaml")
+	cmdResourceExport.Flags().BoolVarP(&o.allNamespaces, "all-namespaces", "A", false, "If present, list ResourceExport across all namespaces")
+	cmdResourceExport.Flags().StringVarP(&o.clusterID, "clusterid", "", "", "List of the ResourceExport of specific clusterID")
+
+	multicluster.GetCmd.AddCommand(cmdResourceExport)
+}
+
+func runEResourceExport(cmd *cobra.Command, args []string) error {
+	optionsResourceExport.validateAndComplete()
+
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+
+	argsNum := len(args)
+	singleResource := false
+	if argsNum > 0 {
+		singleResource = true
+	}
+	var resExports []multiclusterv1alpha1.ResourceExport
+	k8sClient, err := raw.AddSchemeToClientCfg(kubeconfig)
+	if err != nil {
+		return err
+	}
+	var res []multiclusterv1alpha1.ResourceExport
+	if singleResource {
+		resourceExportName := args[0]
+		resourceExport := multiclusterv1alpha1.ResourceExport{}
+		err = k8sClient.Get(context.TODO(), types.NamespacedName{
+			Namespace: optionsResourceExport.namespace,
+			Name:      resourceExportName,
+		}, &resourceExport)
+		if err != nil {
+			return err
+		}
+		gvks, unversioned, err := k8sClient.Scheme().ObjectKinds(&resourceExport)
+		if err != nil {
+			return err
+		}
+		if !unversioned && len(gvks) == 1 {
+			resourceExport.SetGroupVersionKind(gvks[0])
+		}
+		res = append(res, resourceExport)
+	} else {
+		resourceExportList := &multiclusterv1alpha1.ResourceExportList{}
+		err = k8sClient.List(context.TODO(), resourceExportList, &client.ListOptions{Namespace: optionsResourceExport.namespace})
+		if err != nil {
+			return err
+		}
+		res = resourceExportList.Items
+	}
+
+	if optionsResourceExport.clusterID != "" {
+		for _, r := range res {
+			if r.Labels["sourceClusterID"] == optionsResourceExport.clusterID {
```

Comment:
@tnqn Hi, I think I can add the listOption, but seems like this line will be still here, because we also have the Get() function which cannot add labelSelector to it, so do we still want this change?

Reply:
I think it only makes sense to set the clusterID filter when it's a list request, i.e. singleResource is not specified, otherwise the filter is meaningless. For example, if we specify resource name and a filter at the same time with kubectl, it will return error:
```
# kubectl get pods antrea-agent-qdsqw -n kube-system -l app=antrea
error: name cannot be provided when a selector is specified
```

Reply:
Hi @tnqn yes, I use label on purpose since annotation doesn't support selector. I remember that filed in spec don't support selector either. I will add prefix, maybe after v1.6.0 is released.